### PR TITLE
Fix pg upgrade script for foreign tables

### PR DESCRIPTION
**Pull request by agedemenli**
DESCRIPTION: Fixes unexpected error for foreign tables when upgrading pg

Postgres does not support truncate triggers on foreign tables.
This PR adds a filter to make our upgrade script exclude foreign tables when creating truncate triggers.

### CODE REVIEW

File: src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql [hunk 1]
```diff
@@ -104,7 +104,7 @@ BEGIN
     --
     -- register triggers
     --
-    FOR table_name IN SELECT logicalrelid FROM pg_catalog.pg_dist_partition
+    FOR table_name IN SELECT logicalrelid FROM pg_catalog.pg_dist_partition JOIN pg_class ON (logicalrelid = oid) WHERE relkind <> 'f'
```
Comment (onderkalaci) [inline src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql:107]: we'd also need to backport this to 11.0-X.


You'd kind of need another PR like this: https://github.com/citusdata/citus/pull/6037/files

**Comment by marcocitus** [comment]
#6103 introduces 11.0-4, which should be added to the upgrade chain, meaning citus--11.0-3--11.1.sql should be renamed to citus--11.0-4--11.1.sql, and citus--11.1-1--11.0-3.sql should be renamed to citus--11.1-1--11.0-4.sql

**Comment by marcocitus** [comment]
> ...distributed/sql/citus--11.0-3--11.1-1.sql → ...distributed/sql/citus--11.0-3--11.0-4.sql

this seems off, it should become 11.0-4--11.1-1

File: src/backend/distributed/sql/downgrades/citus--11.1-1--11.0-4.sql [hunk 2]
```diff
@@ -86,6 +86,7 @@ DROP FUNCTION pg_catalog.get_all_active_transactions(OUT datid oid, OUT process_
                                                      OUT worker_query BOOL, OUT transaction_number int8, OUT transaction_stamp timestamptz,
                                                      OUT global_pid int8);
 #include "../udfs/get_all_active_transactions/11.0-1.sql"
+#include "../udfs/citus_finish_pg_upgrade/11.0-1.sql"
```
Comment (hanefi) [inline src/backend/distributed/sql/downgrades/citus--11.1-1--11.0-4.sql:89]: The end of this downgrade is 11.0-4 and I expect to see the same definition of citus_finish_pg_upgrade. However, the earlier version of this UDF should be installed on the other downgrade script in here, namely `citus--11.0-4--11.0-3.sql`

File: src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql (hunk 1 above)
Comment (hanefi) [inline src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql:107]: Now that we ship this change in 11.0-4 is this file really necessary?

I am unsure.

Comment (hanefi) [inline src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql:107]: The only difference between `11.1-1.sql` and 11.0-4.sql` is one line

    PERFORM columnar_internal.columnar_ensure_am_depends_catalog();
vs

    PERFORM citus_internal.columnar_ensure_am_depends_catalog();

Is this really expected?

Comment (agedemenli) [reply src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.1-1.sql:107]: This file existed before this PR.
The difference you mentioned was made on #5911

**Comment by marcocitus** [comment]
This looks correct to me (but would be good to have another set of eyes)

**Review by hanefi — APPROVED** [review]
